### ISSUE TITLE
Improve performance.

### ISSUE THREAD

**Issue by mbh95** [mbh95/abstract-art#4]
Major performance issues:

- Lots of WebGL contexts is bad (Chrome won't even let me create more than 16). See https://webgl2fundamentals.org/webgl/lessons/webgl-multiple-views.html for possible solution.
- Re-compiling shaders often is slow. Make sure React doesn't call the recompile effect more than necessary.

**Comment by mbh95** [comment]
Maybe compilation can be done in parallel? Would be nice to have a placeholder loading animation during compilation.

**Comment by mbh95** [comment]
Going to close this. Base performance is greatly improved now that there is only one gl context and I confirmed that glsl recompiling is happening only when it needs to. 

I implemented the one-canvas strategy detailed on webgl2fundamentals.org. Thanks @greggman for an invaluable repository of knowledge and for being so active on StackOverflow :)
